src/redux/books/books.js

```diff
@@ -38,29 +21,76 @@ export default function booksReducer(state = defaultState, action) {
           id: action.id,
           title: action.title,
           author: action.author,
-          progress: action.progress,
-          currentChapter: action.currentChapter,
+          category: action.category,
         },
       ];
     case REMOVE:
       return state.filter((book) => book.id !== action.id);
+    case FETCH:
+      return action.list;
     default:
       return state;
   }
 }
 
-// action creators for add and remove books
+// action creators for fetch, add and remove books
+
+export const fetchBook = (list) => ({ type: FETCH, list });
+
+export const fetchBookAsync = () => async (dispatch) => {
+  await fetch(URL)
+    .then((response) => response.json())
+    .then((books) => {
+      const list = [];
+      Object.keys(books).forEach((key) => {
+        list.push({
+          id: key,
+          title: books[key][0].title,
+          author: books[key][0].author,
+          category: books[key][0].category,
+        });
+      });
+      dispatch(fetchBook(list));
+    });
+};
 
 export const addBook = (payload) => ({
   type: ADD,
   id: payload.id,
   title: payload.title,
   author: payload.author,
-  progress: payload.progress,
-  currentChapter: payload.currentChapter,
+  category: payload.category,
 });
 
-export const removeBook = (payload) => ({
+export const bookFetch = ({
+  id, title, author, category,
+}) => async (dispatch) => {
+  const newBook = {
+    item_id: id,
+    title,
+    author,
+    category,
+  };
+  await fetch(URL, {
+    method: 'POST',
+    body: JSON.stringify(newBook),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  dispatch(addBook(newBook));
+};
+
+export const removeBook = (id) => ({
   type: REMOVE,
-  id: payload.id,
+  id,
 });
+
+export const bookRemove = ({ id }) => async (dispatch) => {
```

Comment:
- [ ] Same here, kindly consider making the changes so your thunk action creators use `createAsyncThunk` instead

Reply:
@Ismailco Same here as above :)